src/meta/rb_map.lean

```diff
@@ -2,41 +2,60 @@
 Copyright (c) 2018 Robert Y. Lewis. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Author: Robert Y. Lewis
-
-Additional operations on native rb_maps and rb_sets.
 -/
 import data.option.defs
 
+/-!
+# rb_map
+
+This file defines additional operations on native rb_maps and rb_sets.
+These structures are defined in core in `init.meta.rb_map`. They are meta objects,
+and are generally the most efficient dictionary structures to use for pure metaprogramming right now.
+-/
+
 namespace native
 namespace rb_set
 
+/-- `filter s P` returns the subset of elements of `s` satisfying `P`. -/
 meta def filter {key} (s : rb_set key) (P : key → bool) : rb_set key :=
 s.fold s (λ a m, if P a then m else m.erase a)
 
+/-- `mfilter s P` returns the subset of elements of `s` satisfying `P`,
+where the check `P` is monadic. -/
 meta def mfilter {m} [monad m] {key} (s : rb_set key) (P : key → m bool) : m (rb_set key) :=
 s.fold (pure s) (λ a m,
   do x ← m,
      mcond (P a) (pure x) (pure $ x.erase a))
 
+/-- `union s t` returns an rb_set containing every element that appears in either `s` or `t`. -/
 meta def union {key} (s t : rb_set key) : rb_set key :=
 s.fold t (λ a t, t.insert a)
 
 end rb_set
 
 namespace rb_map
 
-meta def find_def {α β} (x : β) (m : rb_map α β) (k : α) :=
-(m.find k).get_or_else x
+/-- `find_def default m key` returns the value corresponding to `key` in `m`, if it exists.
+Otherwise it returns `default`. -/
+meta def find_def {α β} (default : β) (m : rb_map α β) (key : α) :=
+(m.find key).get_or_else default
 
-meta def insert_cons {α β} [has_lt α] (k : α) (x : β) (m : rb_map α (list β)) : rb_map α (list β) :=
-m.insert k (x :: m.find_def [] k)
+/-- `insert_cons k x m` adds `x` to the list corresponding to key `k`. -/
+meta def insert_cons {α β} [has_lt α] (key : α) (x : β) (m : rb_map α (list β)) :
```

Comment:
This definition should not be used, and instead we should use `rb_lmap`.
Could you do either of the following:
* Remove this definition
* State in the docstring that this is deprecated and that one should use `rb_lmap` instead?

Reply:
If it's alright, I'll make this change in a separate PR. It's only used once in mathlib and seems easy to replace. But if anyone is using it in an external project, it will be confusing if it suddenly disappears, and this commit message won't indicate that the change was made here.

Reply:
ok